backend/pkg/api/connect/integration/user_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/carlmjohnson/requests"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	v1alpha1 "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1"
+	v1alpha1connect "github.com/redpanda-data/console/backend/pkg/protogen/redpanda/api/dataplane/v1alpha1/dataplanev1alpha1connect"
+)
+
+func (s *APISuite) TestListUsers() {
+	t := s.T()
+	require := require.New(t)
+	assert := assert.New(t)
+
+	t.Run("list users with valid request (connect-go)", func(t *testing.T) {
+		// 1. Create some users in Redpanda
+		ctx, cancel := context.WithTimeout(context.Background(), 12*time.Second)
+		defer cancel()
+
+		// Helper function to create a user
+		createUser := func(username string) {
+			childCtx, cancel := context.WithTimeout(ctx, 3*time.Second)
+			defer cancel()
+
+			err := s.redpandaAdminClient.CreateUser(childCtx, username, "random", adminapi.ScramSha256)
+			require.NoError(err)
+		}
+		deleteUser := func(username string) {
+			childCtx, cancel := context.WithTimeout(ctx, 3*time.Second)
+			defer cancel()
+
+			err := s.redpandaAdminClient.DeleteUser(childCtx, username)
+			assert.NoError(err)
+		}
+
+		username1 := "console-integration-test-list-users-1"
+		username2 := "console-integration-test-list-users-2"
+		createUser(username1)
+		createUser(username2)
+		defer deleteUser(username1)
+		defer deleteUser(username1)
```

Comment:
@weeco this seems to be a bug :)

Reply:
Thanks @krystianity for the unexpected review :D, will fix